File: src/test/java/tech/sirwellington/alchemy/generator/GeolocationGeneratorsTest.java
package tech.sirwellington.alchemy.generator;

import org.junit.jupiter.api.RepeatedTest;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.*;
import static tech.sirwellington.alchemy.generator.GeolocationGenerators.latitudes;
import static tech.sirwellington.alchemy.generator.GeolocationGenerators.longitudes;

class GeolocationGeneratorsTest extends BaseGeneratorTest {

    @RepeatedTest(DEFAULT_ITERATIONS)
    void testLatitudes() {
        AlchemyGenerator<Double> generator = latitudes();
        assertThat(generator, notNullValue());

        for (int i = 0; i < 100; i++) {
            Double latitude = generator.get();
            assertThat(latitude, greaterThanOrEqualTo(-90.0));
            assertThat(latitude, lessThanOrEqualTo(90.0));
        }
    }

    @RepeatedTest(DEFAULT_ITERATIONS)
    void testLongitudes() {
        AlchemyGenerator<Double> generator = longitudes();
        assertThat(generator, notNullValue());

        for (int i = 0; i < 100; i++) {
            Double longitude = generator.get();
            assertThat(longitude, greaterThanOrEqualTo(-180.0));
            assertThat(longitude, lessThanOrEqualTo(180.0));
        }
    }
}

File: src/test/java/tech/sirwellington/alchemy/generator/GeolocationGeneratorsTest.kt
/*
 * Copyright © 2019. Sir Wellington.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 *
 * You may obtain a copy of the License at
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package tech.sirwellington.alchemy.generator

import org.hamcrest.Matchers.greaterThanOrEqualTo
import org.hamcrest.Matchers.lessThanOrEqualTo
import org.hamcrest.Matchers.notNullValue
import org.junit.Assert.assertThat
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.runners.MockitoJUnitRunner

/**

 * @author SirWellington
 */
@RunWith(MockitoJUnitRunner::class)
class GeolocationGeneratorsTest
{

    @Before
    @Throws(Exception::class)
    fun setUp()
    {
    }

    @Test
    fun testLatitudes()
    {
        val generator = GeolocationGenerators.latitudes()
        assertThat(generator, notNullValue())

        doInLoop { i ->
            val latitude = generator.get()

            assertThat(latitude, greaterThanOrEqualTo(-90.0))
            assertThat(latitude, lessThanOrEqualTo(90.0))
        }
    }

    @Test
    fun testLongitudes()
    {
        val generator = GeolocationGenerators.longitudes()
        assertThat(generator, notNullValue())

        doInLoop { i ->
            val latitude = generator.get()

            assertThat(latitude, greaterThanOrEqualTo(-180.0))
            assertThat(latitude, lessThanOrEqualTo(180.0))
        }
    }

}